File: src/main/java/com/apiApp/CateringFacilityAPI/model/jpa/MenuItem.java
package com.apiApp.CateringFacilityAPI.model.jpa;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import javax.persistence.Column;
import javax.persistence.ManyToOne;
import javax.persistence.MappedSuperclass;
import javax.validation.constraints.NotNull;

@MappedSuperclass
public class MenuItem extends BaseEntity {

    @NotNull
    @Column(length = 150)
    private String name;

    @NotNull
    private Double price;

    @Column(length = 5000)
    private String description;

    @NotNull
    private boolean listedInMenu;

    @NotNull
    @ManyToOne
    @JsonIgnore
    private Facility facility;

    public Facility getFacility() {
        return facility;
    }

    @JsonProperty
    public void setFacility(Facility facility) {
        this.facility = facility;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public boolean getListedInMenu() {
        return listedInMenu;
    }

    public void getListedInMenu(boolean listedInMenu) {
        this.listedInMenu = listedInMenu;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: frontend/src/app/model/MenuItem.ts
import {Facility} from "./Facility";
import {BaseEntity} from "./BaseEntity";

export class MenuItem extends BaseEntity{
  constructor(id: number, name: string, price: number, description: string, listedInMenu: boolean, facility: Facility) {
    super(id);
    this.name = name;
    this.price = price;
    this.description = description;
    this.listedInMenu = listedInMenu;
    this.facility = facility;
  }

  public name: string;
  public price: number;
  public description: string;
  public listedInMenu: boolean;
  public facility: Facility;
}
